lightning-invoice/src/lib.rs

```diff
@@ -85,14 +85,16 @@ mod sync;
 
 pub use de::{ParseError, ParseOrSemanticError};
 
-// TODO: fix before 2037 (see rust PR #55527)
-/// Defines the maximum UNIX timestamp that can be represented as `SystemTime`. This is checked by
-/// one of the unit tests, please run them.
-const SYSTEM_TIME_MAX_UNIX_TIMESTAMP: u64 = core::i32::MAX as u64;
+/// The maximum timestamp that can be represented as a [`Duration`] since the UNIX epoch while
+/// allowing for adding an expiry without overflowing.
+const MAX_TIMESTAMP: u64 = core::i64::MAX as u64;
 
-/// Allow the expiry time to be up to one year. Since this reduces the range of possible timestamps
-/// it should be rather low as long as we still have to support 32bit time representations
-const MAX_EXPIRY_TIME: u64 = 60 * 60 * 24 * 356;
+/// The maximum expiry allowed, represented as a [`Duration`] since the invoice timestamp.
```

Comment:
Why have a max expiry at all? Can we just update the expiry-check logic to do a saturating add?

Reply:
Good idea!